### PR TITLE
fix: set up non-default `ipx` providers if `options.ipx` is set

### LINKED ISSUE #1414: IPX provider not working properly when the default provider is not IPX
I have custom provider (thumbor) as my default provider in the config and in some case I still need IPX provider, so I pass provider props in the `NuxtImg` components as `<NuxtImg provider="ipx" />` and the result is 404. `_ipx` path is not found.

The config I had:
```ts
image: {
  ipx: {},
  provider: "thumbor",
  providers: {
    thumbor: { ... }
  }
}
```

When I omit the `ipx` property, it throw `Unknown provider: IPX` error. The config works when I replace the default provider with `ipx`.

Is it expected behavior?

I create a minimal reproduction using `ipx` and `unsplash` provider to suit my problem: https://stackblitz.com/edit/nuxt-image-ipx-repro?file=app.vue

Thank you!

### PR DESCRIPTION
**Pull request by whatcould**
Fixes #1414

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Previously, specifying a custom provider would prevent setup of ipx, which happens in a `'nitro:init` hook. That is, there was no way use ipx if it wasn't your default provider.

Now, you just need to add `ipx` to the image options:

```
image: {
  ipx: {},
  provider: "yourprovider",
  providers: {
    yourprovider: { ... }
  }
}
```

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -134,12 +134,12 @@ ${providers.map(p => `  ['${p.name}']: { provider: ${p.importName}, defaults: ${
...
 
     nuxt.hook('nitro:init', async (nitro) => {
-      if (!options.provider || options.provider === 'ipx' || options.provider === 'ipxStatic') {
+      if (!options.provider || options.provider === 'ipx' || options.provider === 'ipxStatic' || 'ipx' in options || 'ipxStatic' in options) {
         const resolvedProvider = nitro.options.static || options.provider === 'ipxStatic'
           ? 'ipxStatic'
           : nitro.options.node ? 'ipx' : 'none'
 
-        imageOptions.provider = options.provider = resolvedProvider
+        imageOptions.provider = options.provider || resolvedProvider
```
Comment (danielroe) [inline src/module.ts:142]:
```suggestion
        imageOptions.provider = options.provider ||= resolvedProvider
```

**Comment by danielroe** [comment]
Pushed a slight refactor - does that still work for you?

**Comment by whatcould** [comment]
Seems fine? I couldn't quite figure out what ipxStatic did or make it work myself — though it does seem to require the ipx initialization, so if you wanted to use it alongside another default, your change would make it impossible.

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
`ipxStatic` is basically `ipx` but only referencing files that were generated at build time, without any runtime component. And it should work fine - users can provide config in `ipx: {}` that gets passed to the `ipxStatic` provider.

File: src/module.ts [hunk 2]
```diff
@@ -134,13 +134,20 @@ ${providers.map(p => `  ['${p.name}']: { provider: ${p.importName}, defaults: ${
...
 
-        imageOptions.provider = options.provider = resolvedProvider
-        options[resolvedProvider] = options[resolvedProvider] || {}
+        if (!options.provider || options.provider === 'ipx' || options.provider === 'ipxStatic') {
+          imageOptions.provider = options.provider = resolvedProvider
+          options[resolvedProvider] = options[resolvedProvider] || {}
+        }
+
+        // handle the case of `ipx: {}` existing in options, but deploying a static site
+        if (resolvedProvider === 'ipxStatic' && options.provider !== resolvedProvider) {
```
Comment (danielroe) [inline src/module.ts:148]:
```suggestion
        if (resolvedProvider === 'ipxStatic') {
```

File: src/module.ts [hunk 3]
```diff
@@ -134,13 +134,20 @@ ${providers.map(p => `  ['${p.name}']: { provider: ${p.importName}, defaults: ${
...
+          options.ipxStatic ||= options.ipx
```
Comment (danielroe) [inline src/module.ts:149]:
```suggestion
        if (resolvedProvider === 'ipxStatic') {
          options.ipxStatic ||= options.ipx || {}
```

**Comment by IO-Fire** [comment]
Fix #1676

**Pull request merged**